app/src/main/java/com/moviefiend/torian/moviefiend/MainActivity.java

```diff
@@ -1,13 +1,26 @@
 package com.moviefiend.torian.moviefiend;
 
+import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 
-public class MainActivity extends AppCompatActivity {
+import com.moviefiend.torian.moviefiend.network.NowPlayingResponse;
+
+public class MainActivity extends AppCompatActivity implements MovieListFragment.Listener {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        MovieListFragment fragment = (MovieListFragment) getFragmentManager().findFragmentById(R.id.movie_list_fragment);
+        fragment.setListener(this);
```

Comment:
Here is a simple example of how `setListener` can break:

```
class MyAwesomeActivity extends Activity {

  public void onMyAwesomeButtonClicked() {
    MyAwesomeFragment fragment = new MyAwesomeFragment();
    fragment.setListener(this);
    // do a fragment transaction
  }

}

```

If your activity gets destructed (via rotation or via other means) then you're left with a fragment that has no listener which means either NPE will be thrown or worse the user will interact with fragment and nothing will happen (in case the fragment implement null checks for the listener) and so you will not know about this happening to users.

If this was purely a Java design question, I would agree that having an explicit `setListener` method is cleaner.
But it's not, unfortunately Android have a very specific lifecycle considerations that affect the design considerations of the app.
If you introduce a `setListener` method then it's part of the API of the class and other developers may think that they can use it whenever they want, which is not the case as the example just showed.
Which leaves you with two options:

1. Write a long comment on the `setListener` method mentioning the perils of its usage.
2. Avoid the method altogether and go with the `getActivity` or `getTargetFragment` for your listener.

Option (2) seems safer to me.